### PR TITLE
[DOCS] Tensorflow models support in 23.0 update

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Updating docs with info about full support for Tensorflow models.

### CODE REVIEW

File: docs/Documentation/model_introduction.md [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
 
 The approach to fully convert a model is considered the default choice, as it allows the full extent of OpenVINO features. The OpenVINO IR model format is used by other conversion and preparation tools, such as the Post-Training Optimization Tool, for further optimization of the converted model.
 
-Conversion is not required for ONNX and PaddlePaddle models, as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
+Conversion is not required for ONNX, PaddlePaddle, and Tensorflow models, as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
```
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/Documentation/model_introduction.md:23]: As for TensorFlow Frontend, it has limitations to support only `.pb` format. Please refer a document with described limitations of TF FE

File: docs/Extensibility_UG/Intro.md [hunk 2]
```diff
@@ -53,13 +53,13 @@ You might prefer implementing a custom operation class if you already have a gen
 
 Mapping of custom operation is implemented differently, depending on model format used for import. You may choose one of the following:
 
-1. If a model is represented in the ONNX (including models exported from Pytorch in ONNX) or PaddlePaddle formats, then one of the classes from [Frontend Extension API](frontend_extensions.md) should be used. It consists of several classes available in C++ which can be used with the `--extensions` option in Model Optimizer or when a model is imported directly to OpenVINO runtime using the `read_model` method. Python API is also available for runtime model import.
+1. If a model is represented in the ONNX (including models exported from Pytorch in ONNX), PaddlePaddle or Tensorflow formats, then one of the classes from [Frontend Extension API](frontend_extensions.md) should be used. It consists of several classes available in C++ which can be used with the `--extensions` option in Model Optimizer or when a model is imported directly to OpenVINO runtime using the `read_model` method. Python API is also available for runtime model import.
```
Comment (rkazants) [inline docs/Extensibility_UG/Intro.md:56]:
```suggestion
1. If a model is represented in the ONNX (including models exported from Pytorch in ONNX), PaddlePaddle or TensorFlow formats, then one of the classes from [Frontend Extension API](frontend_extensions.md) should be used. It consists of several classes available in C++ which can be used with the `--extensions` option in Model Optimizer or when a model is imported directly to OpenVINO runtime using the `read_model` method. Python API is also available for runtime model import.
```

File: docs/Extensibility_UG/Intro.md [hunk 3]
```diff
@@ -53,13 +53,13 @@ You might prefer implementing a custom operation class if you already have a gen
...
 
-Existing of two approaches simultaneously is explained by two different types of frontends used for model conversion in OpenVINO: new frontends (ONNX, PaddlePaddle) and legacy frontends (TensorFlow, Caffe, Kaldi and Apache MXNet). Model Optimizer can use both front-ends in contrast to the direct import of model with `read_model` method which can use new frontends only. Follow one of the appropriate guides referenced above to implement mappings depending on framework frontend.
+Existing of two approaches simultaneously is explained by two different types of frontends used for model conversion in OpenVINO: new frontends (ONNX, PaddlePaddle and Tensorflow) and legacy frontends (Caffe, Kaldi and Apache MXNet). Model Optimizer can use both front-ends in contrast to the direct import of model with `read_model` method which can use new frontends only. Follow one of the appropriate guides referenced above to implement mappings depending on framework frontend.
```
Comment (rkazants) [inline docs/Extensibility_UG/Intro.md:60]:
```suggestion
Existing of two approaches simultaneously is explained by two different types of frontends used for model conversion in OpenVINO: new frontends (ONNX, PaddlePaddle and TensorFlow) and legacy frontends (Caffe, Kaldi and Apache MXNet). Model Optimizer can use both front-ends in contrast to the direct import of model with `read_model` method which can use new frontends only. Follow one of the appropriate guides referenced above to implement mappings depending on framework frontend.
```

File: docs/Extensibility_UG/frontend_extensions.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 The goal of this chapter is to explain how to use Frontend extension classes to facilitate mapping of custom operations from framework model representation to OpenVINO representation. Refer to [Introduction to OpenVINO Extension](Intro.md) to understand entire flow.
 
-This API is applicable for new frontends only, which exist for ONNX and PaddlePaddle. If a different model format is used, follow legacy [Model Optimizer Extensions](../MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md) guide.
+This API is applicable for new frontends only, which exist for ONNX, PaddlePaddle and Tensorflow. If a different model format is used, follow legacy [Model Optimizer Extensions](../MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md) guide.
```
Comment (rkazants) [inline docs/Extensibility_UG/frontend_extensions.md:5]:
```suggestion
This API is applicable for new frontends only, which exist for ONNX, PaddlePaddle and TensorFlow. If a different model format is used, follow legacy [Model Optimizer Extensions](../MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md) guide.
```

File: docs/Extensibility_UG/frontend_extensions.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ This section covers the case when a single operation in framework representation
 
 5. Each attribute in OpenVINO operation can be initialized from one of the attributes of original operation or by some predefined constant value. Value of copied attributes cannot contain expressions, value is accepted as-is, so type of a value should be compatible.
 
-> **NOTE**: `OpExtension` class is currently available for ONNX frontend only. PaddlePaddle frontend has named inputs and outputs for operation (not indexed) therefore OpExtension mapping is not applicable for this case.
+> **NOTE**: `OpExtension` class is currently available for ONNX and Tensorflow frontends. PaddlePaddle frontend has named inputs and outputs for operation (not indexed) therefore OpExtension mapping is not applicable for this case.
```
Comment (rkazants) [inline docs/Extensibility_UG/frontend_extensions.md:23]:
```suggestion
> **NOTE**: `OpExtension` class is currently available for ONNX and TensorFlow frontends. PaddlePaddle frontend has named inputs and outputs for operation (not indexed) therefore OpExtension mapping is not applicable for this case.
```

File: docs/MO_DG/prepare_model/convert_model/supported_model_formats.md [hunk 6]
```diff
@@ -20,9 +20,9 @@
 
 **OpenVINO IR (Intermediate Representation)** - the proprietary format of OpenVINO™, benefiting from the full extent of its features.
 
-**ONNX, PaddlePaddle** - formats supported directly, which means they can be used with OpenVINO Runtime without any prior conversion. For a guide on how to run inference on ONNX and PaddlePaddle, see how to [Integrate OpenVINO™ with Your Application](../../../OV_Runtime_UG/integrate_with_your_application.md).
+**ONNX, PaddlePaddle, Tensorflow** - formats supported directly, which means they can be used with OpenVINO Runtime without any prior conversion. For a guide on how to run inference on ONNX, PaddlePaddle, or Tensorflow, see how to [Integrate OpenVINO™ with Your Application](../../../OV_Runtime_UG/integrate_with_your_application.md).
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/supported_model_formats.md:23]:
```suggestion
**ONNX, PaddlePaddle, TensorFlow** - formats supported directly, which means they can be used with OpenVINO Runtime without any prior conversion. For a guide on how to run inference on ONNX, PaddlePaddle, or TensorFlow, see how to [Integrate OpenVINO™ with Your Application](../../../OV_Runtime_UG/integrate_with_your_application.md).
```

File: docs/MO_DG/prepare_model/convert_model/supported_model_formats.md [hunk 7]
```diff
@@ -20,9 +20,9 @@
...
 
-**TensorFlow, PyTorch, MXNet, Caffe, Kaldi** - formats supported indirectly, which means they need to be converted to OpenVINO IR before running inference. The conversion is done with Model Optimizer and in some cases may involve intermediate steps.
+**PyTorch, MXNet, Caffe, Kaldi** - formats supported indirectly, which means they need to be converted to OpenVINO IR before running inference. The conversion is done with Model Optimizer and in some cases may involve intermediate steps.
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/supported_model_formats.md:25]: `PyTorch` is invalid to have here.

File: docs/glossary.md [hunk 8]
```diff
@@ -73,13 +73,13 @@ Glossary of terms used in OpenVINO™
 |   OpenVINO™ Core is a software component that manages inference on certain Intel(R) hardware devices: CPU, GPU, GNA, etc.    
 
 | OpenVINO™ API 
-|   The basic default API for all supported devices, which allows you to load a model from Intermediate Representation or convert from ONNX, PaddlePaddle file formars, set input and output formats and execute the model on various devices.
+|   The basic default API for all supported devices, which allows you to load a model from Intermediate Representation or convert from ONNX, PaddlePaddle, Tensorflow file formats, set input and output formats and execute the model on various devices.
```
Comment (rkazants) [inline docs/glossary.md:76]:
```suggestion
|   The basic default API for all supported devices, which allows you to load a model from Intermediate Representation or convert from ONNX, PaddlePaddle, TensorFlow file formats, set input and output formats and execute the model on various devices.
```

File: docs/glossary.md [hunk 9]
```diff
@@ -73,13 +73,13 @@ Glossary of terms used in OpenVINO™
...
 
 | OpenVINO™ Runtime 
 |   A C++ library with a set of classes that you can use in your application to infer input tensors and get the results.
 
 | <code>ov::Model</code> 
-|   A class of the Model that OpenVINO™ Runtime reads from IR or converts from ONNX, PaddlePaddle formats. Consists of model structure, weights and biases.
+|   A class of the Model that OpenVINO™ Runtime reads from IR or converts from ONNX, PaddlePaddle, Tensorflow formats. Consists of model structure, weights and biases.
```
Comment (rkazants) [inline docs/glossary.md:82]:
```suggestion
|   A class of the Model that OpenVINO™ Runtime reads from IR or converts from ONNX, PaddlePaddle, TensorFlow formats. Consists of model structure, weights and biases.
```

File: docs/MO_DG/prepare_model/convert_model/supported_model_formats.md (hunk 7 above)
Comment (sgolebiewski-intel) [inline docs/MO_DG/prepare_model/convert_model/supported_model_formats.md:25]:
```suggestion
**MXNet, Caffe, Kaldi** - formats supported indirectly, which means they need to be converted to OpenVINO IR before running inference. The conversion is done with Model Optimizer and in some cases may involve intermediate steps.
```

Comment (sgolebiewski-intel) [reply docs/MO_DG/prepare_model/convert_model/supported_model_formats.md]: I have removed it in https://github.com/openvinotoolkit/openvino/pull/15974/commits/39a1ee3f08b94bd7a999c9b485d0c53a77ccda54

**Review by kblaszczak-intel — APPROVED** [review]

File: docs/Documentation/model_introduction.md [hunk 10]
```diff
@@ -12,15 +12,15 @@
...
 
 The approach to fully convert a model is considered the default choice, as it allows the full extent of OpenVINO features. The OpenVINO IR model format is used by other conversion and preparation tools, such as the Post-Training Optimization Tool, for further optimization of the converted model.
 
-Conversion is not required for ONNX and PaddlePaddle models, as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
+Conversion is not required for ONNX, PaddlePaddle, and Tensorflow models (check [TensorFlow Frontend Capabilities and Limitations](../resources/tensorflow_frontend.md)), as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
```
Comment (rkazants) [inline docs/Documentation/model_introduction.md:23]:
```suggestion
Conversion is not required for ONNX, PaddlePaddle, and TensorFlow models (check [TensorFlow Frontend Capabilities and Limitations](../resources/tensorflow_frontend.md)), as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
```

**Review by rkazants — APPROVED** [review]
one last comment. Other looks good to me

**Comment by ilya-lavrenov** [comment]
@sgolebiewski-intel @rkazants should we also add similar snippets for TF Lite?
It has `.tflite` file name extension.

**Comment by sgolebiewski-intel** [comment]
> @sgolebiewski-intel @rkazants should we also add similar snippets for TF Lite? It has `.tflite` file name extension.

Sure, I will add it. PR underway :)

**Pull request merged**